dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -849,6 +851,18 @@ protected void updateEPerson(Context context, HttpServletRequest request, EPerso
             ePersonService.setMetadataSingleValue(context, eperson, "eperson", "identifier", "fgs", null, fgs);
         }
 
+        if (idmIds != null && !idmIds.isEmpty()) {
+            idmIds.stream().forEach((idmId) -> {
+                try {
+                    ePersonService.setMetadataSingleValue(
+                        context, eperson, "eperson", "idm", "id", null, idmId.toString()
+                    );
+                } catch (SQLException e) {
+                    log.warn("Could not set idm id of person: " + idmId);
+                }
+            });
+        }
```

Comment:
If my comprehension is good, Shibboleth could return multiple value for IDM-ID header.
And you iterate on each value, to set a **single** metadata into the eperson. Is it correct ? 
In this case, don't loop and set only one IDM-ID (first or last as you want)